### PR TITLE
Resource validation error

### PR DESCRIPTION
**Pull request by SimonRichardson**
The validation of a metadata should also be checked if we're also
checking the model target. The previous code just expected that the
charm is a CAAS charm, but we can't guarantee that at all.

## QA steps

Tests pass.

### CODE REVIEW

File: cmd/juju/application/deployer/deployer_test.go [hunk 1]
```diff
@@ -339,17 +385,6 @@ func (s *deployerSuite) setupMocks(c *gc.C) *gomock.Controller {
 	return ctrl
 }
 
-func (s *deployerSuite) expectResolveCharm(err error, times int) {
```
Comment (SimonRichardson) [inline cmd/juju/application/deployer/deployer_test.go:342]: wasn't used, dead code.

File: cmd/juju/application/deployer/deployer.go [hunk 2]
```diff
@@ -591,9 +593,20 @@ func (d *factory) validateResourcesNeededForLocalDeploy(charmMeta *charm.Meta) e
...
+	if modelType == model.IAAS {
 		return nil
 	}
+
+	// It is expected that everything onwards is a CAAS model and so we should
+	// also check that the metadata is also kubernetes as well. If not then
+	// something is wrong and we're deploying a machine (IAAS) charm on a CAAS
+	// model.
+	if !set.NewStrings(charmMeta.Series...).Contains(series.Kubernetes.String()) && len(charmMeta.Containers) == 0 {
+		return errors.Errorf("expected kubernetes charm metadata, unexpected series or base")
```
Comment (manadart) [inline cmd/juju/application/deployer/deployer.go:607]: "container-based"?

**Review by manadart — APPROVED** [review]
Great.

File: cmd/juju/application/deployer/deployer.go (hunk 2 above)
Comment (SimonRichardson) [reply cmd/juju/application/deployer/deployer.go:607]: Yes, good shout...

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**